### PR TITLE
Add wave equation PINN example (Part 7)

### PR DESCRIPTION
**Pull request by gpartin**
## Summary

Add Part 7 to the PDEs tutorial series: solving the 1D wave equation using physics-informed neural networks (PINNs).

## Motivation

The existing PDE tutorials cover diffusion, Burgers', Laplace, pendulum, and Navier-Stokes equations, but there is no wave equation example. The wave equation is one of the most fundamental PDEs in physics and engineering (acoustics, electromagnetics, vibrations), making it a natural addition to the tutorial series.

## Problem Setup

The example solves the 1D wave equation:

```
u_tt = c^2 * u_xx,  x ∈ [0,1], t ∈ [0,1], c = 1
u(x, 0) = sin(πx)           (initial displacement)
u_t(x, 0) = 0               (initial velocity)
u(0, t) = u(1, t) = 0       (Dirichlet BCs)
```

Exact solution: `u(x,t) = sin(πx) cos(πt)`

## Implementation Details

The wave equation is second-order in time, which requires special handling compared to the first-order diffusion example:

1. **PDE residual**: `f_PINN = d²y/dt² - c² * d²y/dx²` using Neuromancer's symbolic `.grad()` chaining
2. **Initial velocity constraint**: Enforced as a separate loss term `dy/dt(x,0) = 0` on dedicated training points at `t = 0`
3. **Three-part loss**: PDE residual + IC/BC supervised + initial velocity (vs two-part in Part 1)

Architecture: 4 hidden layers of 50 neurons with Tanh activation, trained for 10,000 epochs.

## Files Changed

- `examples/PDEs/Part_7_PINN_WaveEquation.py` — Complete wave equation PINN example following the Part 1-6 pattern
- `examples/PDEs/README.md` — Added Part 7 entry to the tutorial list

## Follows Existing Conventions

- Same code structure as Part 1 (Diffusion): data generation → dataset creation → neural net → symbolic PDE → loss → Trainer
- Same imports: `DictDataset`, `blocks.MLP`, `Node`, `variable`, `PenaltyLoss`, `Problem`, `Trainer`
- Same plotting style with 2D contour + 3D surface views
- Includes analytical solution comparison and relative L2 error computation

### CODE REVIEW

**Comment by drgona** [comment]
@gpartin thank you for your contribution. Can you please create a well-documented Jupyter notebook version of your example? Thank you.
